.github/workflows/build.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script for building fuzzer,benchmark pairs in CI."""
+import sys
+import subprocess
+
+# Don't build php benchmark since it fills up disk in GH actions.
+OSS_FUZZ_BENCHMARKS = [
+    'bloaty_fuzz_target',
+    'curl_curl_fuzzer_http',
+    'irssi_server-fuzz',
+    'jsoncpp_jsoncpp_fuzzer',
+    'libpcap_fuzz_both',
+    'mbedtls_fuzz_dtlsclient',
+    'openssl_x509',
+    'sqlite3_ossfuzz',
+    'systemd_fuzz-link-parser',
+    'zlib_zlib_uncompress_fuzzer',
+]
+
+STANDARD_BENCHMARKS = [
+    'freetype2-2017',
+    'harfbuzz-1.3.2',
+    'lcms-2017-03-21',
+    'libjpeg-turbo-07-2017',
+    'libpng-1.2.56',
+    'libxml2-v2.9.2',
+    'openthread-2019-12-23',
+    'proj4-2017-08-14',
+    're2-2014-12-09',
+    'vorbis-2017-12-11',
+    'woff2-2016-05-06',
+    'wpantund-2018-02-27',
+]
+
+
+def get_build_targets(benchmarks, fuzzer):
+    """Get build targets for |fuzzer| and each benchmark in |benchmarks| to
+    pass to make."""
+    return ['build-%s-%s' % (fuzzer, benchmark) for benchmark in benchmarks]
+
+
+def delete_docker_images():
+    """Delete docker images."""
+    # TODO(metzman): Don't delete base-runner/base-builder so it
+    # doesn't need to be pulled for every target.
+    result = subprocess.run(['docker', 'images', '-q'],
+                            stdout=subprocess.PIPE,
+                            check=True)
+    image_names = result.stdout.splitlines()
+    subprocess.run(['docker', 'rmi', '-f'] + image_names, check=False)
+
+
+def pull_base_images():
+    """Pull base images."""
+    # Pull base-image so that make doesn't rebuild it from scratch.
+    subprocess.run(['docker', 'pull', 'gcr.io/fuzzbench/base-image'],
+                   check=True)
+    subprocess.run(['docker', 'pull', 'gcr.io/fuzzbench/base-builder'],
+                   check=True)
+    subprocess.run(['docker', 'pull', 'gcr.io/fuzzbench/base-runner'],
+                   check=True)
+
+
+def make_builds(build_targets):
+    """Use make to build each target in |build_targets|."""
+    success = True
+    for target in build_targets:
+        pull_base_images()
+        result = subprocess.run(['make', target], check=False)
```

Comment:
This may be worth trying but I will do single thread build for now since most of the docker builds themselves call into things like which themselves are multithreaded.